File: src/types/brief.ts
import type { GitHubBriefData } from "./github";
import type { CalendarEvent } from "./calendar";
import type { EmailHeader } from "./email";

export interface BriefData {
  github: GitHubBriefData | null;
  calendar: CalendarEvent[];
  email: EmailHeader[];
  generated_at: number;
}

File: src-tauri/src/models/brief.rs
use serde::{Deserialize, Serialize};
use super::{GitHubBriefData, CalendarEvent, EmailHeader};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BriefData {
    pub github: Option<GitHubBriefData>,
    pub calendar: Vec<CalendarEvent>,
    pub email: Vec<EmailHeader>,
    pub generated_at: i64,
}

impl Default for BriefData {
    fn default() -> Self {
        Self {
            github: None,
            calendar: Vec::new(),
            email: Vec::new(),
            generated_at: chrono::Utc::now().timestamp(),
        }
    }
}
